src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts

```diff
@@ -0,0 +1,258 @@
+      )
+      .subscribe((daemonStatsArray) => {
+        this.rgwEndpoints.value = daemonStatsArray.map((daemonStats) => {
+          const protocol = daemonStats.frontendConfig.includes('ssl_port') ? 'https' : 'http';
+          return `${protocol}://${daemonStats.hostname}:${daemonStats.port}`;
+        });
+        const options: SelectOption[] = this.rgwEndpoints.value.map(
+          (endpoint: string) => new SelectOption(false, endpoint, '')
+        );
+        this.rgwEndpoints.options = [...options];
+      });
+
+    this.multiClusterService.getCluster().subscribe((clusters) => {
+      this.clusterDetailsArray = Object.values(clusters['config'])
+        .flat()
+        .filter((cluster) => cluster['cluster_alias'] !== 'local-cluster');
+      this.isMultiClusterConfigured = this.clusterDetailsArray.length > 0;
+      if (!this.isMultiClusterConfigured) {
+        this.stepTitles = ['Create Realm & Zonegroup', 'Create Zone', 'Export Multi-site token'];
+      } else {
+        this.selectedCluster = this.clusterDetailsArray[0]['name'];
+      }
+    });
+  }
+
+  createForm() {
+    this.multisiteSetupForm = new CdFormGroup({
+      realmName: new UntypedFormControl('default_realm', {
+        validators: [Validators.required]
+      }),
+      zonegroupName: new UntypedFormControl('default_zonegroup', {
+        validators: [Validators.required]
+      }),
+      zonegroup_endpoints: new UntypedFormControl(null, [Validators.required]),
+      zoneName: new UntypedFormControl('default_zone', {
+        validators: [Validators.required]
+      }),
+      zone_endpoints: new UntypedFormControl(null, {
+        validators: [Validators.required]
+      }),
+      username: new UntypedFormControl('default_system_user', {
+        validators: [Validators.required]
+      }),
+      cluster: new UntypedFormControl(null, {
+        validators: [Validators.required]
+      })
+    });
+
+    if (!this.isMultiClusterConfigured) {
+      this.exportTokenForm = new CdFormGroup({});
+    }
+  }
+
+  showSubmitButtonLabel() {
+    if (this.isMultiClusterConfigured) {
+      return !this.wizardStepsService.isLastStep()
+        ? this.actionLabels.NEXT
+        : $localize`Configure Multi-site`;
+    } else {
+      return !this.wizardStepsService.isLastStep() ? this.actionLabels.NEXT : $localize`Close`;
+    }
+  }
+
+  showCancelButtonLabel() {
+    return !this.wizardStepsService.isFirstStep()
+      ? this.actionLabels.BACK
+      : this.actionLabels.CANCEL;
+  }
+
+  onNextStep() {
+    if (!this.wizardStepsService.isLastStep()) {
+      this.wizardStepsService.getCurrentStep().subscribe((step: WizardStepModel) => {
+        this.currentStep = step;
+      });
+      if (this.currentStep.stepIndex === 2 && !this.isMultiClusterConfigured) {
+        this.loading = true;
+        this.onSubmit();
+      } else {
+        this.wizardStepsService.moveToNextStep();
+      }
+    } else {
+      this.loading = true;
+      this.onSubmit();
```

Comment:
This means it loads on submit, no? You could add it there

```suggestion
        this.onSubmit();
      } else {
        this.wizardStepsService.moveToNextStep();
      }
    } else {
      this.onSubmit();
```

Reply:
Done, Thanks